dfd-wrapper/src/test/java/com/revolut/rxdata/dfd_wrapper/TestHelpers.kt

```diff
@@ -0,0 +1,122 @@
+package com.revolut.rxdata.dfd_wrapper
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.TimeoutCancellationException
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withTimeout
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+import kotlin.time.Duration
+
+/*
+ * Copyright (C) 2023 Revolut
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// inspired by Kotest
+infix fun <T, U : T> T.shouldBe(expected: U?) = assertEquals(expected, this)
+
+fun Job.assertNotComplete() {
+    assertFalse(isCompleted)
+}
+
+fun Job.assertComplete() {
+    assertTrue(isCompleted)
+}
+
+fun <T> Flow<T>.firstOrError() = flow { emit(first()) }
+
+fun <T> Flow<T>.test(scope: CoroutineScope): TestObserver<T> {
+    return TestObserver(this, scope)
+}
+
+// inspired by https://proandroiddev.com/from-rxjava-to-kotlin-flow-testing-42f1641d8433
+class TestObserver<T>(
```

Comment:
we already have flow turbine testing lib which should cover for all this

Reply:
Yes, this should work fine if we don't need to port existing DOD tests.

BTW, I tried to update Turbine to latest version (0.12.1), but then some flow-extensions tests fail with `expected: java.lang.IllegalStateException@fd0e5b6<java.lang.IllegalStateException> but was: java.lang.IllegalStateException@4eed46ee<java.lang.IllegalStateException>`.

Seems a typical case of [stacktrace recovery](https://github.com/Kotlin/kotlinx.coroutines/blob/master/docs/topics/debugging.md#stacktrace-recovery), but that existed for a long time. Probably something has changed in Turbine threading. 🤔

Reply:
Updated Turbine to 0.8.0, because with 0.7.0 my `Flow.test()` calls failed. 🤷‍♂️
Tests in `flow-extensions` still pass with new version.